### PR TITLE
Add manual page showing file locations

### PR DESCRIPTION
**Pull request by benbc**

### CODE REVIEW

**Comment by benbc** [comment]
@jjaderberg Can you give me some help with this, please? In particular:
- how do we describe "where ever you happen to have unpacked the zip/tarball"? (I currently have `<neo4j-root>`)
- how should we format paths with variables in (e.g. `%APPDATA%\Neo4j Community Edition\neo4j.conf` and those with `<neo4j-root>` if they survive the question above)
- What's the right way to refer to a Mac? We are inconsistent at the moment.

And, do you generally like the approach I'm taking? We'll add other files as we do the work.

File: enterprise/ha/src/docs/dev/arbiter.asciidoc [hunk 1]
```diff
@@ -9,8 +9,8 @@ Instead, what is called arbiter instances can be deployed.
 They can be regarded as cluster participants in that their role is to take part in master elections with the single purpose of breaking ties in the election process.
 That makes possible a scenario where you have a cluster of 2 Neo4j database instances and an additional arbiter instance and still enjoy tolerance of a single failure of either of the 3 instances.
 
-Arbiter instances are configured in the same way as Neo4j HA members are -- through the _conf/neo4j.conf_ file.
-The instance is configured to be an arbiter by setting the +org.neo4j.server.database.mode+ option in _conf/neo4j.conf_ to +ARBITER+.
+Arbiter instances are configured in _neo4j.conf_ using the same settings as standard Neo4j HA members.
+The instance is configured to be an arbiter by setting the +org.neo4j.server.database.mode+ option in _neo4j.conf_ to +ARBITER+.
```
Comment (jjaderberg) [inline enterprise/ha/src/docs/dev/arbiter.asciidoc]: Second reference to the config file is unnecessary, `in _neo4j.conf_` can be removed. Please substitute backtick for plus around the setting name and value in this sentence.

File: manual/server-examples/src/docs/dev/server-unmanaged-extensions.asciidoc [hunk 2]
```diff
@@ -54,7 +54,7 @@ The full source code is found here:
 https://github.com/neo4j/neo4j/blob/{neo4j-git-tag}/community/server-examples/src/main/java/org/neo4j/examples/server/unmanaged/HelloWorldResource.java[HelloWorldResource.java]
 
 Having built your code, the resulting jar file (and any custom dependencies) should be placed in the +$NEO4J_SERVER_HOME/plugins+ directory.
-We also need to tell Neo4j where to look for the extension by adding some configuration in the _conf/neo4j.conf_ file:
+We also need to tell Neo4j where to look for the extension by adding some configuration in _neo4j.conf_:
```
Comment (jjaderberg) [inline manual/server-examples/src/docs/dev/server-unmanaged-extensions.asciidoc:59]: Will future work also touch the _plugins_ directory? We need to change `+$NEO4J_SERVER_HOME/plugins+` as well, but I'm not sure to what.

**Comment by jjaderberg** [comment]
@benbc I like the approach.
- Let's use the same formatting for the whole path also when it includes placeholders like `$HOME`, `%APPDATA%` and `<neo4j-root>`.
- I'm not fond of `<neo4j-root>` because the notation is afaik non-standard and the placeholder not explicitly defined. I don't have a better suggestion so let's use it until we come up with something better.
- Not sure how consistent or not we are about referring to _a Mac_. On the File location page we are talking about operating systems, not about machines, correct? Then the correct phrase would be "Mac OS X", or "OS X" for short.

It is good that on some pages the first reference to the config file is a link to the File locations page.

Please separate the cross reference name and target with a space after the comma: `<<file-locations, _neo4j.conf_>>`.

File: manual/server-examples/src/docs/dev/server-unmanaged-extensions.asciidoc (hunk 2 above)
Comment (benbc) [inline manual/server-examples/src/docs/dev/server-unmanaged-extensions.asciidoc:59]: Yes, we'll make sure that location is documented like _conf_ and references to it normalized.

**Comment by benbc** [comment]
@jjaderberg I've addressed your comments.

**Comment by benbc** [comment]
@srbaker FYI -- here is the new page showing where files live. You'll need to add your stuff here.

**Comment by benbc** [comment]
@jjaderberg Are you happy to merge this?

**Pull request merged**